petastorm/spark/spark_dataset_converter.py

```diff
@@ -139,19 +158,55 @@ def __len__(self):
         """
         return self.dataset_size
 
-    def make_tf_dataset(self):
+    def make_tf_dataset(
+            self,
+            batch_size=32,
+            prefetch=None,
+            num_epochs=None,
+            workers_count=4,
+            **petastorm_reader_kwargs
+    ):
         """
         Make a tensorflow dataset.
 
         This method will do the following two steps:
           1) Open a petastorm reader on the materialized dataset dir.
           2) Create a tensorflow dataset based on the reader created in (1)
 
+        :param batch_size: The number of items to return per batch
+        :param prefetch: Prefetch size for tensorflow dataset. If None will use
+            tensorflow autotune size. Note only available on tensorflow>=1.14
+        :param num_epochs: An epoch is a single pass over all rows in the dataset.
+            Setting ``num_epochs`` to ``None`` will result in an infinite number
+            of epochs.
+        :param workers_count: An int for the number of workers to use in the
+            reader pool. This only is used for the thread or process pool. Default value 4.
+        :param petastorm_reader_kwargs: arguments for `petastorm.make_batch_reader()`,
+            exclude these arguments: "dataset_url", "num_epochs", "workers_count".
+
         :return: a context manager for a `tf.data.Dataset` object.
                  when exit the returned context manager, the reader
                  will be closed.
         """
-        return TFDatasetContextManager(self.cache_dir_url)
+
+        # override some arguments default values of petastorm reader
+        petastorm_reader_kwargs['num_epochs'] = num_epochs
+        petastorm_reader_kwargs['workers_count'] = workers_count
+
+        env_rank, env_size = get_env_rank_and_size()
```

Comment:
* move it to a utility function since it will be used by both TF and torch
* `env_rank` -> rank